mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster.go

```diff
@@ -2883,6 +2883,10 @@ func flattenSecurityConfig(d *schema.ResourceData, sc *dataproc.SecurityConfig)
 	if sc == nil {
 		return nil
 	}
+	// API side change means that sc can be non-nil but contain only nils
+	if sc.IdentityConfig == nil && sc.KerberosConfig == nil {
+		return nil
+	}
```

Comment:
Might be better focusing on only sc.KerberosConfig == nil, as that's the only one used. Might fall through if IdentityConfig is non nil but sc.KerberosConfig is

Reply:
Yea, I think we may want to just make `flattenKerberosConfig` handle nil values passed in. Looking at the API logic, there seems to be multiple combinations that could be sent down (including a new field that hasn't been made visible yet), and it might be easier to maintain one level down.

Reply:
Given the tests have stopped running on the PR I'll push a fix, and once the current TC build ends we can trigger a more scoped-down build that runs only the relevant tests